protocol/bin/js-code-generators.js

```diff
@@ -17,15 +17,15 @@ const channelMapPrelude = `function _channelMapRpcHelper(channelConfig: ChannelC
 }
 `
 
-function rpcChannelMap (name, callbackType, innerParamType, responseType) {
-  return `export function ${name}RpcChannelMap (channelConfig: ChannelConfig<*>, request: $Exact<${['requestCommon', callbackType, innerParamType].filter(t => t).join(' & ')}>): ChannelMap<*> {
-  return _channelMapRpcHelper(channelConfig, (incomingCallMap, callback) => ${name}Rpc({...request, incomingCallMap, callback}))
+function rpcChannelMap (methodName, name, callbackType, innerParamType, responseType) {
+  return `\nexport function ${name}RpcChannelMap (channelConfig: ChannelConfig<*>, request: $Exact<${['requestCommon', callbackType, innerParamType].filter(t => t).join(' & ')}>): ChannelMap<*> {
+  return _channelMapRpcHelper(channelConfig, (incomingCallMap, callback) => { engineRpcOutgoing(${methodName}, request, callback, incomingCallMap) })
```

Comment:
call engine helper directly instead of through the other rpc helper (and we might deprecate the non promise ones anyways)